docs/use_bls_wallet_dapp.md

```diff
@@ -0,0 +1,109 @@
+
+Below are the instructions for 2 ways you can add Quill to your browser.
+
+### Prebuilt
+
+Go to the [releases page](https://github.com/web3well/bls-wallet/releases) and scroll down to the latest release. In the `Assets` section, download the extension for either Chrome, Firefox, or Opera. To install, simply drag and drop the file into your browser on the extensions page or follow instructions for installing extensions from a file for your browser.
+
+### From Repo
+
+Follow the instructions in either [Local Development](./local_development.md) or [Remote Development](./remote_development.md) to setup this repo and install Quill.
+
+### Setup Quill
+
+Folow the setup instructions when ou first open Quill, and then create a wallet if there is not one there already.
+
+## Connect Your dApp to Quill
+
+Next, connect your dApp to Quill just like you would any other extension wallet.
+
+```typescript
+const provider = providers.Web3Provider(window.ethereum);
+
+await window.ethereum.request({ method: "eth_accounts" });
+```
+
+## Send Multi-Action Transaction
+
+Finally, you can populate & send your multi-action transaction. In the following example, we will do an approve & swap with a DEX (decentralized exchange) in one transaction.
+
+### Check that window.ethereum Supports Multi-Action Transactions
+
+Since any browser extension wallet could be used, make sure that it is Quill before allowing a multi-action transaction.
+
+```typescript
+const areMultiActionTransactionSupported = "isQuill" in window.ethereum && window.ethereum.isQuill;
+// Branch here depending on the result
+if (areMultiActionTransactionSupported) {
+  ...
+}
+```
+
+### Populate the Transactions (Actions)
+
+First, we will populate the transactions (actions) we want to send.
+
+```typescript
+// Connect to the contracts and get the signer.
+const erc20Contract = new ethers.Contract(erc20Address, erc20Abi, provider);
+const dexContract = new ethers.Contract(dexAddress, dex20Abi, provider);
+
+const signer = provider.getSigner();
+
+// Populate the token approval transaction.
+const approveTransaction = await erc20Contract
+  .connect(signer)
+  .populateTransaction.approve(dexAddress, amount);
+
+// Populate the token swap transaction.
+const swapTransaction = await dexContract
+  .connect(signer)
+  .populateTransaction.swap(erc20Address, amount, otherERC20Address);
+```
+
+### Send the Transaction
+
+Then, send the populatated transactions.
+
+```typescript
+const transactionHash = await window.ethereum.request({
+  method: "eth_sendTransaction",
+  // Note that any number of transactions can be bundled & sent.
+  params: [approveTransaction, swapTransaction],
```

Comment:
I think adding an explicit footnote that "this is experimental and a bit of a deviation from the standard" would be more informative to the devs so they can be mindful when using other web3 libraries like ethersJS.